packages/block-library/src/gallery/style.scss

```diff
@@ -83,23 +84,41 @@
 	// Responsive fallback value, 2 columns
 	& .blocks-gallery-image,
 	& .blocks-gallery-item {
-		width: calc(100% / 2 - 16px);
+		width: calc((100% / 2) - (#{ $grid-size-large } * 2));
 	}
 
 	&.columns-1 .blocks-gallery-image,
 	&.columns-1 .blocks-gallery-item {
-		width: calc(100% / 1 - 16px);
+		width: calc(100% / 1);
 	}
 
 	@include break-small {
 		@for $i from 3 through 8 {
 			&.columns-#{ $i } .blocks-gallery-image,
 			&.columns-#{ $i } .blocks-gallery-item {
-				width: calc(100% / #{ $i } - 16px);
+				width: calc((100% / #{ $i }) - (#{ $grid-size-large } * #{ $i }));
 			}
 		}
 	}
 
+	// Unset right margins.
+	@for $j from 1 through 8 {
```

Comment:
Why 1 through 8?

Reply:
The other for loop starts at 3 because 1 and 2 have their own explicit rules for mobile, and 3 columns and up is for above mobile breakpoints anyway.

But the rules to remove the right margin on the rightmost tile applies across the spectrum, i.e. from 1 up to and including 8 columns.

Reply:
Might be good to add a comment then, stating that 8 columns is the max or something because that wasn't clear to me 😄